pandas/tests/scalar/interval/test_arithmetic.py

```diff
@@ -0,0 +1,54 @@
+from datetime import timedelta
+
+import numpy as np
+import pytest
+
+from pandas import Interval, Timedelta, Timestamp
+
+
+@pytest.mark.parametrize("method", ["__add__", "__sub__"])
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timestamp_interval_add_subtract_timedelta(method, delta):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(
+        Timestamp("2017-01-01 00:00:00"), Timestamp("2018-01-01 00:00:00")
+    )
+
+    result = getattr(interval, method)(delta)
+
+    left = getattr(interval.left, method)(delta)
+    right = getattr(interval.right, method)(delta)
+    expected = Interval(left, right)
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timedelta_add_timestamp_interval(delta):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(
+        Timestamp("2017-01-01 00:00:00"), Timestamp("2018-01-01 00:00:00")
+    )
+
+    result = delta + interval
+
+    left = interval.left + delta
+    right = interval.right + delta
+    expected = Interval(left, right)
+
+    assert result == expected
+
+
+@pytest.mark.parametrize("method", ["__add__", "__sub__"])
+def test_numeric_interval_add_subtract_timedelta_raises(method):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(1, 2)
+    delta = Timedelta(days=7)
```

Comment:
can you parametrize over delta like in the previous test?  (id probably not parametrize over `__add__` and `__sub__`, but your call)

some more intervals that might be worth checking
- float dtype
- float nan
- datetime NaT
- timedelta NaT

Reply:
the raising should also work in the other direction, i.e. `Interval(Timestamp) + numeric` should fail

Reply:
Updated with some more cases